packages/hdwallet-native/src/crypto/isolation/adapters/bitcoin.ts

```diff
@@ -57,22 +64,12 @@ export class ECPairAdapter implements SignerAsync, ECPairInterfaceAsync {
     return isolatedKey.ecdhRaw.bind(isolatedKey);
   }
 
-  async sign(hash: bcrypto.NonDigest | bcrypto.Digest<"hash256">, lowR?: boolean): Promise<Buffer> {
-    assertType(ByteArray(), hash);
-
+  async sign(hash: Uint8Array, lowR?: boolean): Promise<Buffer> {
+    assertType(ByteArray(32), hash);
     lowR = lowR ?? this.lowR;
-    const sig = await (async () => {
-      if (!hash.algorithm) {
-        assertType(ByteArray(32), hash);
-        return !lowR
-          ? await this._isolatedKey.ecdsaSign(null, hash)
-          : await SecP256K1.Signature.signCanonically(this._isolatedKey, null, hash);
-      } else {
-        return !lowR
-          ? await this._isolatedKey.ecdsaSign(hash.algorithm, hash.preimage)
-          : await SecP256K1.Signature.signCanonically(this._isolatedKey, hash.algorithm, hash.preimage);
-      }
-    })();
+    const sig = !lowR
```

Comment:
Diff looks scary but is actually the same as before, just simplified now to conform to new `sign` interface